client/rest-high-level/src/main/java/org/elasticsearch/client/ilm/LifecyclePolicy.java

```diff
@@ -80,7 +80,10 @@ public LifecyclePolicy(String name, Map<String, Phase> phases) {
             if (ALLOWED_ACTIONS.containsKey(phase.getName()) == false) {
                 throw new IllegalArgumentException("Lifecycle does not support phase [" + phase.getName() + "]");
             }
-            phase.getActions().forEach((actionName, action) -> {
+           if (phase.getName().equals("delete") && phase.getActions().size() == 0) {
+               throw new IllegalArgumentException("phase [" + phase.getName() + "] must contain actions");
```

Comment:
In order to keep the messaging around phase and actions consistent with the below validation, can you please change the message to "must define actions"?

 ```suggestion
               throw new IllegalArgumentException("phase [" + phase.getName() + "] must define actions");
```